docs/js-core-1/week-2/lesson.md

```diff
@@ -382,253 +383,35 @@ Write a function, similar to the last exercise, that:
-2. What is printed to the console?
-3. Have you learned anything new during this exercise?
-
-```js
-const daysOfWeek = [
-  "Monday",
-  "Tuesday",
-  "Wednesday",
-  "Thursday",
-  "Friday",
-  "Saturday",
-  "Sunday",
-];
-
-function workingDay(day) {
-  return day + " is a working day";
-}
-
-function weekendDay(day) {
-  return day + " is at the weekend!";
-}
+### for of loop
 
-for (let i = 0; i < daysOfWeek.length; i++) {
-  if (i < 5) {
-    let day = workingDay(daysOfWeek[i]);
-    console.log(day);
-  } else {
-    let day = weekendDay(daysOfWeek[i]);
-    console.log(day);
-  }
-}
-```
-
-:::
+[mdn &rarr;](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of)
 
-:::note Exercise
+The `for of` loop The for...of statement creates a loop iterating over iterable objects, including: built-in String, Array, array-like objects (e.g., arguments or NodeList), TypedArray, Map, Set, and user-defined iterables. It invokes a custom iteration hook with statements to be executed for the value of each distinct property of the object.
 
-### Playing computer II
+The `for` loop syntax has special places for each of those three things. Here's the same loop as the first while loop above, as a for loop:
 
-1. Working in pairs or groups, you have to predict the output of this program without executing it.
-2. What is printed to the console?
-3. Have you learned anything new during this exercise?
+![For loop](https://user-images.githubusercontent.com/31692/75388870-9213a880-58dd-11ea-90e6-4e67eabf390b.png)
 
 ```js
-const a = 4;
-const b = 8;
-
-const multiplyNumbers = function (a, b) {
-  return a * b;
-};
-
-const addNumbers = function (a, b, c) {
-  return a + b + c;
-};
-
-for (let i = 0; i < 5; ++i) {
-  if (i < 3) {
-    const day = addNumbers(i, 2, a);
-    console.log(day);
-  } else {
-    const day = multiplyNumbers(i, 4);
-    console.log(day);
-  }
+for (let i = 0; i < 20; i++) {
+  someFunction(i);
 }
 ```
 
-:::
+### Exercise (10 minutes)
 
 :::note Exercise
 
-### Playing computer III
-
-1. Again, working in pairs or groups, you have to predict the output of this program without executing it.
-2. What is printed to the console?
-3. What was difficult about this exercise?
-4. Have you learned anything new?
+Write a function, similar to the last exercise, that:
 
-```js
-let x = 2;
-let y = 4;
-let a = 2;
-let b = 20;
-
-const f1 = function (a, b) {
-  return a * b;
-};
-
-const f2 = function (a, b, c) {
-  return a + b + c;
-};
-
-console.log(x);
-x = 3;
-y = f1(x, 2);
-console.log(y);
-
-for (let i = 0; i < 10; ++i) {
-  a = a + 1;
-  if (i % 2 === 0) {
-    const d = f2(i, y, a);
-    console.log(d);
-  } else {
-    const e = f1(i, 2);
-    console.log(e);
-  }
-}
-```
+- Takes one number `n` as a parameter
+- Adds all numbers from `0` to `n`. For example, if the input is `3`, the output should be `0 + 1 + 2 + 3`
+- You should use a for loop
```

Comment:
I can't comment on the lines because they're not changed, but I think ~all of the glossary below is now obsolete for this lesson?

Reply:
resolved by moving to https://github.com/CodeYourFuture/syllabus/pull/458/commits/03b1c2e8eb1120f4b0eb0bfb41e2d30fae1f67f1